### PR TITLE
Bump zwave-js to 12.0.2 and add `soft_reset` capability

### PR DESCRIPTION
**Pull request by raman325**
- [Z-Wave JS 12.0.1](https://github.com/zwave-js/node-zwave-js/releases/tag/12.0.1)
- [Z-Wave JS 12.0.2](https://github.com/zwave-js/node-zwave-js/releases/tag/12.0.2)

Also adds a new option to control the soft-reset behavior for 500 series controllers. Also extends the timeout for killing the process from 3 to 30 seconds which should help with corrupt caches.

@AlCalzone

### CODE REVIEW

File: zwave_js/rootfs/etc/cont-init.d/config.sh [hunk 1]
```diff
@@ -105,20 +105,22 @@ host_chassis=$(bashio::host.chassis)
 if [ "${host_chassis}" == "vm" ]; then
     soft_reset=false
     bashio::log.info "Virtual Machine detected, disabling soft-reset"
+elif bashio::config.true 'disable_soft_reset'; then
```
Comment (MartinHjelmare) [inline zwave_js/rootfs/etc/cont-init.d/config.sh:108]: Should we allow the user to override the automatic detection completely and enable soft-reset also on VM?

Comment (AlCalzone) [reply zwave_js/rootfs/etc/cont-init.d/config.sh:108]: I think so, yes. If they manage to configure the VM correctly, enabling soft-reset is good.

Comment (raman325) [reply zwave_js/rootfs/etc/cont-init.d/config.sh]: OK so I have now changed it to a pick list of three options -> automatic, enabled, or disabled (defaults to automatic)

File: zwave_js/translations/en.yaml [hunk 2]
```diff
@@ -37,5 +37,11 @@ configuration:
     description: >-
       This must be provided in order to include devices with the S2
       Unauthenticated security class.
+  disable_soft_reset:
```
Comment (MartinHjelmare) [inline zwave_js/translations/en.yaml:40]: This needs to be updated.

File: zwave_js/DOCS.md [hunk 3]
```diff
@@ -145,13 +145,10 @@ It will not be able to control any real devices.
...
-on startup or when certain commands are issued to the controller. Soft-resets
-can be problematic in certain environments, so Z-Wave JS has an option to
-disable this functionality. This option tells the add-on how to handle soft-resets as follows:
-1. Automatic - the add-on will decide whether soft-reset should be enabled or disabled. This is the deafult option and should work for most people.
-2. Enabled - Soft-reset will be explicitly be enabled.
-3. Disabled - Soft-reset will be explicitly be disabled.
+This setting tells the add-on how to handle soft-resets for 500 series controllers:
+1. Automatic - the add-on will decide whether soft-reset should be enabled or disabled for 500 series controllers. This is the deafult option and should work for most people.
+2. Enabled - Soft-reset will be explicitly enabled for 500 series controllers.
+3. Disabled - Soft-reset will be explicitly disabledfor 500 series controllers.
```
Comment (MartinHjelmare) [inline zwave_js/DOCS.md]:
```suggestion
3. Disabled - Soft-reset will be explicitly disabled for 500 series controllers.
```

**Review by MartinHjelmare — APPROVED** [review]

**Review by MartinHjelmare — APPROVED** [review]

**Pull request merged**